### PR TITLE
Use simple physics engine

### PR DESCRIPTION
**Pull request by tobiasbp**
* Move player in all directions with arrow keys.
* Use [simple physics engine](https://api.arcade.academy/en/latest/api/physics_engines.html#arcade-physicsenginesimple) for movement and collisions with walls.
* Remove arguments to _Player_ init restricting x position, as movement is now restricted by the physics engine.

### CODE REVIEW

File: my_game.py [hunk 1]
```diff
@@ -165,21 +169,28 @@ def on_update(self, delta_time):
...
+        elif self.down_pressed and not self.up_pressed:
+            self.player.change_y = -PLAYER_SPEED
 
         # Move player with joystick if present
         if self.joystick:
-            self.player.change_x = round(self.joystick.x) * PLAYER_SPEED_X
+            self.player.change_x = round(self.joystick.x) * PLAYER_SPEED
 
-        # Update player sprite
-        self.player.on_update(delta_time)
```
Comment (ivan-BRHM) [inline my_game.py:182]: Perhaps we could try adding back 
`self.player.on_update`

**Review by ivan-BRHM — CHANGES_REQUESTED** [review]
We should be able to move both vertically and horisontally simultaneously.

File: my_game.py (hunk 1 above)
Comment (tobiasbp) [reply my_game.py:182]: > Perhaps we could try adding back `self.player.on_update`

Updating is now done by the physics engine (PE). We run the [_PE.update()_ function](https://api.arcade.academy/en/latest/_modules/arcade/physics_engines.html#PhysicsEngineSimple.update). That, in turn, calls __move_sprite()_ to manipulate the sprite's position.  You can read the source code [here](https://api.arcade.academy/en/latest/_modules/arcade/physics_engines.html).

So, no update function on the _Sprite_ is in use.

**Comment by tobiasbp** [comment]
> We should be able to move both vertically and horisontally simultaneously.

Diagonal movement for the player is now possible. Let me know what you think.

File: my_game.py (hunk 1 above)
Comment (ivan-BRHM) [reply my_game.py:182]: ah, i see that - that's very good

**Comment by ivan-BRHM** [comment]
This looks good. But currently we move much faster when moving diagonally - because both our change_x and change_y is equal to PLAYER_SPEED.
I personally think that the _sum of the two_, should equal PLAYER_SPEED

File: my_game.py [hunk 2]
```diff
@@ -165,21 +169,31 @@ def on_update(self, delta_time):
...
+            self.player.change_x = -PLAYER_SPEED
         elif self.right_pressed and not self.left_pressed:
-            self.player.change_x = PLAYER_SPEED_X
+            self.player.change_x = PLAYER_SPEED
+        else:
+            self.player.change_x = 0
+
+        # Up/down
+        if self.up_pressed and not self.down_pressed:
+            self.player.change_y = PLAYER_SPEED
```
Comment (ivan-BRHM) [inline my_game.py:183]: Currently, change_x and change_y can equal PLAYER_SPEED - this means we move much faster diagonally.
I think the _sum of the two_ should equal PLAYER_SPEED when we are moving.

**Review by ivan-BRHM — CHANGES_REQUESTED** [review]
Looks good! A small change is requested

File: my_game.py (hunk 2 above)
Comment (tobiasbp) [reply my_game.py:182]: I have reverted back to movement only in x and y. A later PR can add diagonal movement. This PR is about the physics engine.

**Pull request merged**